build_tools/hack/build_env_diag.py

```diff
@@ -0,0 +1,882 @@
+#!/usr/bin/env python
+
+# 
+#   Contributed by TaiXeflar and Scott Todd
+#   TheRock Project building system pre-build diagnosis script
+#   License follows TheRock project
+#
+#   Variables: "FULL_UPPER_VARIABLE" are been called to CMake-Like style.
+#       *CMake                  *Python
+#       WIN32 -->               WINDOWS
+#       Linux -->               LINUX
+#   CMAKE_MAJOR_VERSION
+# 
+#   !  Hint: This script doesn't raise/throw back warnings/errors.
+#   This script is for detecting environments use, We do a global scan on all requirements at once.
+#   We do not want users have to fix its environment one-by-one and get frustrated, so the diagnosis won't throw errors on it.
+#   If Users(Yes, It's you! maybe.) running this script have throwback errors, It must have bug in it.
+#       PLZ! Please report it as new issue or open in a new disscus <3
+#
+
+import os, platform, subprocess, re, time
+from typing import Literal
+from shutil import disk_usage, which
+import warnings
+from os import environ as ENV    
```

Comment:
Style-wise I'd prefer to use this pattern instead:

```python
import os

os.getenv("ENVIRONMENT_VARIABLE_NAME_1")
os.getenv("ENVIRONMENT_VARIABLE_NAME_2")
```

Sticking to the same style across the project makes code refactoring easier, while hiding behind shorthand like this `ENV` makes it harder. We don't gain too much by saving a few extra characters.